homeassistant/components/sipcall/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""The SIP Call integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_HASS_CONFIG, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.NOTIFY]
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Discord component."""
+
+    hass.data[DATA_HASS_CONFIG] = config
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SIP Call from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    # We simply store the config data. The actual connection will be made by notify
+    hass.data[DOMAIN][entry.entry_id] = entry.data
+
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass, Platform.NOTIFY, DOMAIN, dict(entry.data), hass.data[DATA_HASS_CONFIG]
+        )
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
```

Comment:
I don't we're unloading anything so we can remove this function

Reply:
We are popping the entry configuration from the config dict. This is to prevent a memory leak. I think it makes sense to keep it

Reply:
I wrote this with the understanding of that we should not need the data stored in hass
Data

Reply:
Removed